In [ ]:
import pandas as pd

In [ ]:
nba_url='https://raw.githubusercontent.com/data602sps/datasetspractice/main/nba.csv'
df1 = pd.read_csv(nba_url)

In [15]:
df1

Name            Team  Number Position   Age Height  Weight  \
0    Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   
1      Jae Crowder  Boston Celtics    99.0       SF  25.0    6-6   235.0   
2     John Holland  Boston Celtics    30.0       SG  27.0    6-5   205.0   
3      R.J. Hunter  Boston Celtics    28.0       SG  22.0    6-5   185.0   
4    Jonas Jerebko  Boston Celtics     8.0       PF  29.0   6-10   231.0   
..             ...             ...     ...      ...   ...    ...     ...   
453   Shelvin Mack       Utah Jazz     8.0       PG  26.0    6-3   203.0   
454      Raul Neto       Utah Jazz    25.0       PG  24.0    6-1   179.0   
455   Tibor Pleiss       Utah Jazz    21.0        C  26.0    7-3   256.0   
456    Jeff Withey       Utah Jazz    24.0        C  26.0    7-0   231.0   
457            NaN             NaN     NaN      NaN   NaN    NaN     NaN   

               College     Salary  
0                Texas  7730337.0  
1            Marquette  6796117.0  
2    Boston University        NaN  
3        Georgia State  1148640.0  
4                  NaN  5000000.0  
..                 ...        ...  
453             Butler  2433333.0  
454                NaN   900000.0  
455                NaN  2900000.0  
456             Kansas   947276.0  
457                NaN        NaN  

[458 rows x 9 columns]

458 rows and 9 columns

In [16]:
print(df1.shape)
print(df1.dtypes)

(458, 9)
Name         object
Team         object
Number      float64
Position     object
Age         float64
Height       object
Weight      float64
College      object
Salary      float64
dtype: object


5 object types, 4 float64 types

In [17]:
df1.describe()

Number         Age      Weight        Salary
count  457.000000  457.000000  457.000000  4.460000e+02
mean    17.678337   26.938731  221.522976  4.842684e+06
std     15.966090    4.404016   26.368343  5.229238e+06
min      0.000000   19.000000  161.000000  3.088800e+04
25%      5.000000   24.000000  200.000000  1.044792e+06
50%     13.000000   26.000000  220.000000  2.839073e+06
75%     25.000000   30.000000  240.000000  6.500000e+06
max     99.000000   40.000000  307.000000  2.500000e+07

I see there are descriptive stats for just the numerical columns. According to the documentation for describe(), you need to specifiy the `include` parameter to see all of the columns.

I see the weight has the highest spread out of the numerical columns.

In [ ]:
df1.head()

This shows the first 5 rows in the dataset.

In [ ]:
df1.tail()

This shows the last 5 rows in the dataset.

2nd dataset

In [ ]:
weather_url='https://raw.githubusercontent.com/data602sps/datasetspractice/main/weather.csv'
df2 = pd.read_csv(weather_url)

In [18]:
df2

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
0              1    EWR  2013    1.0   1.0   0.0  37.04  21.92  53.97   
1              2    EWR  2013    1.0   1.0   1.0  37.04  21.92  53.97   
2              3    EWR  2013    1.0   1.0   2.0  37.94  21.92  52.09   
3              4    EWR  2013    1.0   1.0   3.0  37.94  23.00  54.51   
4              5    EWR  2013    1.0   1.0   4.0  37.94  24.08  57.04   
...          ...    ...   ...    ...   ...   ...    ...    ...    ...   
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
0        230.0    10.35702  11.918651     0.0    1013.9   10.0  
1        230.0    13.80936  15.891535     0.0    1013.0   10.0  
2        230.0    12.65858  14.567241     0.0    1012.6   10.0  
3        230.0    13.80936  15.891535     0.0    1012.7   10.0  
4        240.0    14.96014  17.215830     0.0    1012.8   10.0  
...        ...         ...        ...     ...       ...    ...  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0  

[8719 rows x 15 columns]

8719 rows and 15 columns

In [ ]:
print(df2.shape)
print(df2.dtypes)

In [19]:
df2.describe()

Unnamed: 0    year        month          day         hour  \
count  8719.000000  8719.0  8718.000000  8718.000000  8718.000000   
mean   4360.000000  2013.0     6.506538    15.679628    11.515715   
std    2517.102832     0.0     3.439961     8.768250     6.916846   
min       1.000000  2013.0     1.000000     1.000000     0.000000   
25%    2180.500000  2013.0     4.000000     8.000000     6.000000   
50%    4360.000000  2013.0     7.000000    16.000000    12.000000   
75%    6539.500000  2013.0     9.000000    23.000000    18.000000   
max    8719.000000  2013.0    12.000000    31.000000    23.000000   

              temp         dewp        humid     wind_dir   wind_speed  \
count  8718.000000  8718.000000  8718.000000  8486.000000  8718.000000   
mean     55.484942    41.794184    62.863333   193.194674     9.335732   
std      18.345817    19.619776    19.572699   110.541333    12.328261   
min      10.940000    -9.040000    13.950000     0.000000     0.000000   
25%      39.920000    26.060000    47.602500   110.000000     5.753900   
50%      55.940000    42.800000    62.330000   220.000000     9.206240   
75%      71.060000    57.920000    78.790000   280.000000    12.658580   
max     100.040000    75.920000   100.000000   360.000000  1048.360580   

         wind_gust       precip     pressure        visib  
count  8718.000000  8719.000000  7780.000000  8719.000000  
mean     10.743374     0.003035  1017.826465     9.224522  
std      14.187117     0.022686     7.396722     2.094044  
min       0.000000     0.000000   983.900000     0.120000  
25%       6.621473     0.000000  1012.800000    10.000000  
50%      10.594357     0.000000  1017.500000    10.000000  
75%      14.567241     0.000000  1022.900000    10.000000  
max    1206.432388     1.180000  1041.900000    10.000000

I see the `count` differs for some of the columns. For example, year has 8719 values and month has 8718. Maybe this is an indication that there are missing values.

In [ ]:
df2.head()

In [ ]:
df2.tail()

3rd dataset

In [21]:
df3 = pd.read_csv('https://raw.githubusercontent.com/djunga/602/main/spotify_artist_data.csv')

In [22]:
df3

Unnamed: 0    Artist Name    Lead Streams           Feats Tracks  \
0              1          Drake  50,162,292,808  19,246,513,666    262   
1              2      Bad Bunny  44,369,032,140   5,391,990,975    163   
2              3     Ed Sheeran  38,153,682,361   2,791,278,201    240   
3              4     The Weeknd  34,767,779,741   4,288,903,657    186   
4              5   Taylor Swift  32,596,728,109     424,053,296    323   
...          ...            ...             ...             ...    ...   
1004        1005          Gotye   1,424,591,618      10,576,378     46   
1005        1006           NIKI   1,422,920,850     231,405,433     55   
1006        1007    Duke Dumont   1,422,096,182      24,622,115    121   
1007        1008  Laura Pausini   1,421,622,604     293,291,443    434   
1008        1009           Beck   1,420,597,420      83,871,063    300   

     One Billion 100 Million Last Updated  
0              6         130     19.09.22  
1              5         118     20.09.22  
2             10          62     19.09.22  
3              8          72     19.09.22  
4              1          96     20.09.22  
...          ...         ...          ...  
1004           1           1     31.07.22  
1005           0           4     14.09.22  
1006           0           2     31.07.22  
1007           0           0     13.09.22  
1008           0           1     17.08.22  

[1009 rows x 8 columns]

1009 rows and 8 columns

In [23]:
print(df3.shape)
print(df3.dtypes)

(1009, 8)
Unnamed: 0       int64
Artist Name     object
Lead Streams    object
Feats           object
Tracks          object
One Billion     object
100 Million     object
Last Updated    object
dtype: object


Besides the Unamed:0 column, each column is of the `object` datatype.

In [24]:
df3.describe()

Unnamed: 0
count  1009.000000
mean    505.000000
std     291.417513
min       1.000000
25%     253.000000
50%     505.000000
75%     757.000000
max    1009.000000

There's only 1 column because by default, describe() only acknowledges numerical columns.

In [25]:
df3.head()

Unnamed: 0   Artist Name    Lead Streams           Feats Tracks  \
0           1         Drake  50,162,292,808  19,246,513,666    262   
1           2     Bad Bunny  44,369,032,140   5,391,990,975    163   
2           3    Ed Sheeran  38,153,682,361   2,791,278,201    240   
3           4    The Weeknd  34,767,779,741   4,288,903,657    186   
4           5  Taylor Swift  32,596,728,109     424,053,296    323   

  One Billion 100 Million Last Updated  
0           6         130     19.09.22  
1           5         118     20.09.22  
2          10          62     19.09.22  
3           8          72     19.09.22  
4           1          96     20.09.22

We can see that there is more than just 1 numerical column, as thought before. We can probably coerce some of these columns to be numerical.

In [ ]:
df3.tail()

4th dataset

In [26]:
df4 = pd.read_csv('https://raw.githubusercontent.com/djunga/602/main/googleplaystore.csv')

In [27]:
df4

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[10841 rows x 13 columns]

10841 rows, 13 columns

In [28]:
print(df4.shape)
print(df4.dtypes)

(10841, 13)
App                object
Category           object
Rating            float64
Reviews            object
Size               object
Installs           object
Type               object
Price              object
Content Rating     object
Genres             object
Last Updated       object
Current Ver        object
Android Ver        object
dtype: object


Mostly `object` columns.

In [29]:
df4.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

We only see 1 numerical column. Maybe there are others that can be coerced, as in the last dataset.

In [30]:
df4.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

`Reviews`, and `Price` can probably be coerced.

In [ ]:
df4.tail()